### PR TITLE
Add Linux uninstall support

### PR DESCRIPTION
**Pull request by gromgit**
Basically three differences:
1. Parameterize default Homebrew path
2. GNU find vs. BSD find
3. .DS_Store not applicable in Linux

### CODE REVIEW

**Comment by dawidd6** [comment]
You can also tweak the CI config to run the uninstall on Linux too.

File: uninstall [hunk 1]
```diff
@@ -16,6 +16,9 @@ options = {
 # global status to indicate whether there is anything wrong.
 @failed = false
 
+@is_mac = RUBY_PLATFORM.to_s.downcase.include? "darwin"
```
Comment (MikeMcQuaid) [inline uninstall]:
```suggestion
@is_mac = !RUBY_PLATFORM[/darwin/].nil?
```

File: uninstall [hunk 2]
```diff
@@ -224,7 +227,13 @@ paths = %w[Frameworks bin etc include lib opt sbin share var]
...
         .map(&:to_s)
 if paths.any?
-  args = %w[-E] + paths + %w[-regex .*/info/([^.][^/]*\.info|dir)]
+  if @is_mac
+    # BSD find
+    args = %w[-E] + paths + %w[-regex .*/info/([^.][^/]*\.info|dir)]
+  else
+    # GNU find
+    args = paths + %w[-regextype posix-extended -regex .*/info/([^.][^/]*\.info|dir)]
+  end
```
Comment (MikeMcQuaid) [inline uninstall]:
```suggestion
  args = if @is_mac
    # BSD find
    %w[-E] + paths + %w[-regex .*/info/([^.][^/]*\.info|dir)]
  else
    # GNU find
    paths + %w[-regextype posix-extended -regex .*/info/([^.][^/]*\.info|dir)]
  end
```

File: uninstall [hunk 3]
```diff
@@ -272,7 +281,7 @@ paths = %w[bin etc include lib opt sbin share var
         .map { |p| HOMEBREW_PREFIX/p }
         .select(&:exist?)
         .map(&:to_s)
-if paths.any?
+if @is_mac && paths.any?
   args = paths + %w[-name .DS_Store]
```
Comment (MikeMcQuaid) [inline uninstall]:
```suggestion
if paths.any?
  args = paths
  args += %w[-name .DS_Store] if @is_mac 
```

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -46,8 +46,7 @@ jobs:
           echo "::add-path::/home/linuxbrew/.linuxbrew/bin:/usr/bin:/bin"
         fi
 
-    - name: Test uninstaling GitHub Actions macOS Homebrew
-      if: runner.os == 'macOS'
```
Comment (dawidd6) [inline .github/workflows/tests.yml]: Leave this line. It's for uninstalling Homebrew which is preinstalled on macOS but not on Ubuntu.

File: uninstall (hunk 3 above)
Comment (gromgit) [reply uninstall]: I can see now that there's a bug in my original change, but shouldn't the logic in this section be:

```
if paths.any?

  # Delete .DS_Store on macOS only...
  if @is_mac
    args = paths + %w[-name .DS_Store]
    if options[:dry_run]
      args << "-print"
    else
      args << "-delete"
    end
    puts "Would delete:" if options[:dry_run]
    sudo "/usr/bin/find", *args
  end

  # ...but remove all empty subdirs on all OSes
  args = paths + %w[-depth -type d -empty]
  if options[:dry_run]
    args << "-print"
  else
    args += %w[-exec rmdir {} ;]
  end
  puts "Would remove directories:" if options[:dry_run]
  sudo "/usr/bin/find", *args

end
```

File: .github/workflows/tests.yml (hunk 4 above)
Comment (gromgit) [reply .github/workflows/tests.yml]: Noted. Will revert in a subsequent commit (just noticed a speling error in the original).

Comment (MikeMcQuaid) [reply .github/workflows/tests.yml]: My bad! If you could fix those it'd be great ❤️

File: uninstall (hunk 3 above)
Comment (MikeMcQuaid) [reply uninstall]: Yup, rereading it again: I agree with your newly suggested version 👍

**Comment by gromgit** [comment]
Any other issues? If not, I can force-push a squashed commit, then PR the bash version. Thanks much!

**Comment by MikeMcQuaid** [comment]
Looks great, thanks again @gromgit!

**Pull request merged**